python/lsst/ts/salobj/domain.py

```diff
@@ -274,45 +273,44 @@ def remove_salinfo(self, salinfo):
         except KeyError:
             return False
 
+    def basic_close(self):
+        """A synchronous and less thorough version of `close`.
+
+        Intended for exit handlers and constructor error handlers.
+        """
+        try:
+            while self._salinfo_set:
+                salinfo = self._salinfo_set.pop()
+                salinfo.basic_close()
+        finally:
+            self.participant.close()
+        self.participant.close()
```

Comment:
Isn't this calling `self.participant.close()` twice? once in `finally` and another here if there's no exception? You can probably remove this one no?

Reply:
Good catch. I'll delete the second one.